airflow/docker_image.go

```diff
@@ -398,16 +398,34 @@ func (d *DockerImage) Push(remoteImage, username, token string) error {
 		// if it does not work with the go library use bash to run docker commands. Support for (old?) versions of Colima
 		err = pushWithBash(&authConfig, remoteImage)
 		if err != nil {
-			return err
+			return "", err
+		}
+	}
+
+	// Get the digest of the pushed image
+	remoteDigest := ""
+	if getRemoteShaTag {
+		out := &bytes.Buffer{}
+		err = cmdExec(containerRuntime, out, nil, "inspect", "--format={{index .RepoDigests 0}}", remoteImage)
+		if err != nil {
+			return remoteDigest, fmt.Errorf("failed to get digest for image %s: %w", remoteImage, err)
+		}
+		// Parse and clean the output
+		digestOutput := strings.TrimSpace(out.String())
+		if digestOutput != "" {
+			parts := strings.Split(digestOutput, "@")
+			if len(parts) == 2 {
+				remoteDigest = parts[1] // Extract the digest part (after '@')
+			}
```

Comment:
@neel-astro Actually this is tightly coupled to push because we are getting the digest that we receive from remote registry response after pushing the image. It's not the local SHA of the newly built image.

Reply:
It is not. Correct me if I am wrong, but here we are executing a separate Docker command to fetch the sha information (`docker inspect`). That operation needs to run on the same underlying image and not necessarily the remote image (the SHA tag would remain the same irrespective of the image tag being used to fetch it). But it's a different operation if you think from `ImageHandler` pov.

I am just trying to limit the `Push` method from getting into an operation it didn't sign up for. It already has a tag/untag that shouldn't be part of it if you go by the method definition.